### PR TITLE
Added suggested changes

### PR DESCRIPTION
**Pull request by do-ruth**
I tested it again with the new changes and everything still works. Right now, I know the user can still input string and cause my program to crash but I will just focus on the other aspects at the moment. I hope everything still looks right to you!

@quietHamster

### CODE REVIEW

**Comment by do-ruth** [comment]
Question:
I feel like my code looks messy. Could I have made separate classes and returned values? I haven't done that in a while and am not familiar with it though.

**Comment by quietHamster** [comment]
good job! I feel great with what you have done! 

these are my tests, I did not attempt any edge cases, like input string, let take care of them later in another project.
we focus on the algorithm on this one.

Test 1: Simple - PASSED
```
Welcome to Ruth's Simple Random Number Generator!
Note, your bounds will be included in the range
Please input the lower bound of your range (i.e. 3)
3
The lower bound is: 3
Please input the upper bound of your range (i.e. 10)
10
The upper bound is: 10
How many random numbers would you like?
1
Generating 1 number(s):
8
```

Test 2: Simple Validation - PASSED
```
Welcome to Ruth's Simple Random Number Generator!
Note, your bounds will be included in the range
Please input the lower bound of your range (i.e. 3)
3
The lower bound is: 3
Please input the upper bound of your range (i.e. 10)
3
Your upper bound is equal to your lower bound. Please input a valid integer.Please input the upper bound of your range (i.e. 10)
4
The upper bound is: 4
How many random numbers would you like?
1
Generating 1 number(s):
3
```

Test 3: Simple Validation - PASSED
```
Welcome to Ruth's Simple Random Number Generator!
Note, your bounds will be included in the range
Please input the lower bound of your range (i.e. 3)
3
The lower bound is: 3
Please input the upper bound of your range (i.e. 10)
-1
Your upper bound is less than your lower bound. Please input a valid integer.Please input the upper bound of your range (i.e. 10)
4
The upper bound is: 4
How many random numbers would you like?
1
Generating 1 number(s):
3
```

Test 4: Algorithm - CONFUSION (not fail though)
```
Welcome to Ruth's Simple Random Number Generator!
Note, your bounds will be included in the range
Please input the lower bound of your range (i.e. 3)
3
The lower bound is: 3
Please input the upper bound of your range (i.e. 10)
10
The upper bound is: 10
How many random numbers would you like?
10
Your range does not fit the range. Enter in an amount smaller than the range.
9
Your range does not fit the range. Enter in an amount smaller than the range.
8
Generating 8 number(s):
9
4
10
8
7
9
9
4
```
I think the number of random numbers that will be generated are independent from the upper bound. 
We have several scenarios here:
- Range 3 to 10: what is our expectance criteria? 3 and 10 are inclusive or exclusive? from my testing I think both are inclusive, i.e. the code will generate 10, 3 as the outcome. Let's say we confirm that the possible unique outcomes are: 3,4,5,6,7,8,9,10 total of 8 outcomes. 
- Your current validation say that, if user input any number greater than 8, it won't go. And, it is not wrong, just not clear! What do you mean by limiting 8 here? Does it mean 8 unique outcomes because the number of possible unique outcomes? If yes, we have some repeat number in my test case. If no, we don't have to care about this, right? It can provide whatever the number of outcomes if we don't care if they are unique or not.
- So, the thing here is just what we define as done criteria!
- I need YOU to tell me your decision how to move this confusion test case forward. Not discussing in the code yet, just the behavior of the app for now

**Comment by do-ruth** [comment]
Thank you for your input! At first, I wanted unique numbers but looking at it now, I'd rather just keep it simple and allow repeated numbers. That means I won't place a limit on the amount of random numbers the user wants.

**Comment by quietHamster** [comment]
Great decision.
Love this _"At first, I wanted unique numbers but looking at it now, I'd rather just keep it simple and allow repeated numbers"_

**Comment by quietHamster** [comment]
Now I will jump to the code

File: src/RandNumGenerator.java [hunk 1]
```diff
@@ -10,27 +10,72 @@ public static void main (String[] args) {
...
-        int amtNum = mrScanner.nextInt();
-        System.out.println("Generating " + amtNum + " number(s):");
-
-        int i = amtNum;
-        while (i > 0) {
-            int randNum = lowerNum + rand.nextInt(upperNum - lowerNum);
-            System.out.println(randNum);
-            i--;
+
+        int neg = 1;
```
Comment (quietHamster) [inline src/RandNumGenerator.java]: I see you are using this variable as a `yes/no flag.`
Have you thought about `boolean`? The best practice is not to "waste" your CPU resource. When we use `int` the resource will be allocate a lot more, while `boolean` only need 2 states `true/false`

File: src/RandNumGenerator.java [hunk 2]
```diff
@@ -10,27 +10,72 @@ public static void main (String[] args) {
...
+            lowerNum = mrScanner.nextInt();
+            if (lowerNum < 0) {
+                System.out.println("You entered a negative number, please input a valid lower bound.");
+            }
+            else {
+                neg = 0;
+                System.out.println("The lower bound is: " + lowerNum);
+
+                //getting upper bound of range
+                int invalNum = 1;
```
Comment (quietHamster) [inline src/RandNumGenerator.java]: Same concern as with `int neg` comment above

File: src/RandNumGenerator.java [hunk 3]
```diff
@@ -10,27 +10,72 @@ public static void main (String[] args) {
...
+                int upperNum;
+                while (invalNum != 0) {
+                    System.out.println("Please input the upper bound of your range (i.e. 10)");
+                    upperNum = mrScanner.nextInt();
+                    if (upperNum < lowerNum) {
+                        System.out.print("Your upper bound is less than your lower bound. Please input a valid integer.");
+                    }
+                    else if (upperNum == lowerNum) {
+                        System.out.print("Your upper bound is equal to your lower bound. Please input a valid integer.");
+                    }
```
Comment (quietHamster) [inline src/RandNumGenerator.java:37]: what do you think if you can combine these `if ... else if ...` into one `if ...`?

File: src/RandNumGenerator.java [hunk 4]
```diff
@@ -10,27 +10,72 @@ public static void main (String[] args) {
...
+                    else {
+                        invalNum = 0;
+                        System.out.println("The upper bound is: " + upperNum);
+
+
+                        //getting amount of numbers
+                        System.out.println("How many random numbers would you like?");
+                        int j = 1;
```
Comment (quietHamster) [inline src/RandNumGenerator.java]: `Variable naming convention`: always have meaningful names because others will have to look at your code, for this simple app, I understand what you mean, but if the app has 1K lines of code, there is no way I can determine. 
Also, consider to use `boolean` here, because you use it as a flag

File: src/RandNumGenerator.java [hunk 5]
```diff
@@ -10,27 +10,72 @@ public static void main (String[] args) {
...
+                        while (j != 0) {
+                            int amtNum = mrScanner.nextInt();
+
+                            int range = upperNum - lowerNum + 1;
+                            if (amtNum > range) {
```
Comment (quietHamster) [inline src/RandNumGenerator.java]: This condition need to be revised if you allow repeat numbers
